src/main/kotlin/app/revanced/patcher/apk/Archive.kt

```diff
@@ -0,0 +1,172 @@
+package app.revanced.patcher.apk
+
+import app.revanced.patcher.logging.Logger
+import app.revanced.patcher.util.xml.LazyXMLInputSource
+import com.reandroid.apk.ApkModule
+import com.reandroid.archive.ByteInputSource
+import com.reandroid.archive.InputSource
+import com.reandroid.arsc.chunk.xml.AndroidManifestBlock
+import com.reandroid.arsc.chunk.xml.ResXmlDocument
+import com.reandroid.xml.XMLDocument
+import com.reandroid.xml.XMLException
+import lanchon.multidexlib2.BasicDexEntry
+import lanchon.multidexlib2.RawDexIO
+import org.jf.dexlib2.dexbacked.DexBackedDexFile
+import org.jf.dexlib2.iface.MultiDexContainer
+import org.jf.dexlib2.writer.io.MemoryDataStore
+import java.io.File
+import java.io.IOException
+import java.util.zip.ZipEntry
+
+private fun isResXml(inputSource: InputSource) = inputSource.openStream().use { ResXmlDocument.isResXmlBlock(it) }
+
+/**
+ * A class for reading/writing files in an [ApkModule].
+ *
+ * @param module The [ApkModule] to operate on.
+ */
+internal class Archive(private val module: ApkModule) : MultiDexContainer<DexBackedDexFile> {
```

Comment:
This class does not belong in ReVanced Patcher because it contains a lot of lower level functions for arsclib. Please create a separate module that wraps around arsclib and use it in ReVanced Patcher. Leave functions that are suitable here in this module though